### PR TITLE
Implement new release banner - Closes #2144

### LINKED ISSUE #2144: Implement the new release banner
### Description
On a second step of #2016, we can Implement the banner design [here](https://projects.invisionapp.com/d/main#/console/17570736/369054984/preview), the release details popup appear when the user clicks on it.
On this step we should use a React popup instead of the implementing outside the React app within a new window.
This means that when there's a new version, we send the raw information through IPC channel,  and we show the banner. when the user clicks on the banner, we open a new modal/popup to show the entire release report. this report is the raw information received using IPC. 

There will be a follow up ticket #2317 to implement the new release dialog

### PR DESCRIPTION
**Pull request by massao**
### What issue have I solved?

#2144 

### How have I implemented/fixed it?

- [x] Refactor flashMessage toolbox component to be more generic;
- [x] Created base header to render signInHeader or topBar component;
- [x] Created flashMessageHolder to be easier to render new flash messages;
- [x] Pass data from the IPC to new release banner;

The release summary is fetched from the release note and consists of the content of the first html tag.

The following items will be done in a followup ticket #2317 .
- [ ] Open modal with IPC data through release banner;
- [ ] Start update through click on dialog;
- [ ] Show toaster that update started;

### How has this been tested?

Change version on package.json, build the application an run with electront o see that the banner shows up.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

File: src/components/toolbox/flashMessage/flashMessage.js [hunk 1]
```diff
@@ -1,79 +1,54 @@
...
-  iconName: '',
-  linkCaption: '',
-  linkClassName: '',
-  linkUrl: '',
+  shouldShow: false,
 };
 
+FlashMessage.displayName = 'FlashMessage';
+FlashMessage.Text = Text;
+FlashMessage.Button = Button;
```
Comment (slaweet) [inline src/components/toolbox/flashMessage/flashMessage.js]: Awesome 🏆

**Comment by massao** [comment]
> * 🐛 The release notes from Github can contain HTML that is escaped (see screenshot)

I'm checking into it.

> * 🐛 In the release notes, the first line is the title of the first section (usually "Fixed bugs" or "Implemented enhancements"), It should IMO show the second line. (see "body" in https://api.github.com/repos/LiskHQ/lisk-hub/releases)

By what I discussed with @reyraa for future releases we are going to have the first line being a heading with the summary. But we can check other alternatives with him, if needed.

> * 🐛 "Read more" seems slightly vertically miss-aligned with the text.
>   Screenshot 2019-08-05 at 10 05 56

This seems to be because the button has a `font-size: 15px;` and the text has `font-size: 13px`,  because it's using `display: flex; align-items: center;` so it should be ok.

> * ❓ On the dark splash screen it looks a bit weird, maybe bottom border would help. Please check with design team.
>   Screenshot 2019-08-05 at 10 05 23

Will check and update this comment with the conclusion.

**Comment by slaweet** [comment]
- :bug: Account initialization is not displayed anymore (I saw it in the HTML, but CSS makes it not visible somehow)

Screenshot 2019-08-05 at 10 56 18

**Comment by slaweet** [comment]
> > * 🐛 In the release notes, the first line is the title of the first section (usually "Fixed bugs" or "Implemented enhancements"), It should IMO show the second line. (see "body" in https://api.github.com/repos/LiskHQ/lisk-hub/releases)
> 
> By what I discussed with @reyraa for future releases we are going to have the first line being a heading with the summary. But we can check other alternatives with him, if needed.

The first line with summary sounds good. Will there be some formatting, like bold or H1-6 @reyraa ? IMO it makes sense to have the summary somehow highlighted on Github as one of the most important parts. Can we already do it for the latest release?
https://github.com/LiskHQ/lisk-hub/releases/tag/v1.19.1 So that we can test what does it look like now and not after the release.

File: src/utils/newRelease.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import FlashMessageHolder from '../components/toolbox/flashMessage/holder';
+import NewReleaseMessage from '../components/newReleaseMessage/newReleaseMessage';
+
+export default {
+  init: () => {
+    const { ipc } = window;
+    if (!ipc) return;
+
+    ipc.on('update:available', (action, { version, releaseNotes }) => {
+      /* istanbul ignore next */
```
Comment (slaweet) [inline src/utils/newRelease.js]: Why? Add a test or a comment here, please.

File: src/components/newReleaseMessage/newReleaseMessage.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  releaseNotes,
+  onClick,
+  ...props
+}) => (
+  <FlashMessage shouldShow {...props}>
+    <FlashMessage.Content>
+      <React.Fragment>
+        <strong>{t('Lisk Hub {{version}}', { version })}</strong>
+        {t(' is out. ')}
+        {releaseNotes.match(regex.releaseSummary)[2]}
```
Comment (slaweet) [inline src/components/newReleaseMessage/newReleaseMessage.js]: IMO it's better to have the component more presentational, by having the logic to extract `releaseSummary` in `src/utils/newRelease.js` and accept only `releaseSummary` prop.

File: src/components/newReleaseMessage/newReleaseMessage.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          {t('Read more')}
+        </TertiaryButton>
+      </React.Fragment>
+    </FlashMessage.Content>
+  </FlashMessage>
+);
+
+NewReleaseMessage.propTypes = {
+  version: PropTypes.string.isRequired,
+  releaseNotes: PropTypes.string.isRequired,
```
Comment (slaweet) [inline src/components/newReleaseMessage/newReleaseMessage.js]: `releaseNotes` is not used/needed anymore now that you have `releaseSummary`

**Comment by reyraa** [comment]
@slaweet @massao Yes, let's use a heading for the summary so it's easier read and we can grab the summary easily.

File: src/utils/newRelease.js (hunk 2 above)
Comment (massao) [reply src/utils/newRelease.js]: Added while developing and forgot there 🤦‍♂

**Comment by reyraa** [comment]
@massao I jazzed up this release note so you can test easily.
https://github.com/LiskHQ/lisk-hub/releases/tag/v1.19.1
Let me know if it works for you.

**Comment by slaweet** [comment]
- :bug: Currently, it displays only the "Release goal" text. Do we want to change the code or the release notes format?

Screenshot 2019-08-05 at 15 39 29

**Comment by slaweet** [comment]
When I tried to change it to display the actual text, I found out :bug: it doesn't wrap the text on multiple lines if too long (or small screen)
Screenshot 2019-08-05 at 15 42 45

File: src/utils/htmlParser.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React, { Fragment, createElement } from 'react';
+import regex from './regex';
+
+const htmlParser = (html = '') => {
```
Comment (slaweet) [inline src/utils/htmlParser.js]: I was confused by the name `htmlParser` at first. IMO `htmlStringToReact` would be more descriptive.

Comment (massao) [reply src/utils/htmlParser.js]: Renamed as suggested.

**Comment by massao** [comment]
> 🐛 Currently, it displays only the "Release goal" text. Do we want to change the code or the release notes format?

For now I set the releaseSummary as being the first `h4` on the release notes.

> When I tried to change it to display the actual text, I found out 🐛 it doesn't wrap the text on multiple lines if too long (or small screen)

Fixed the wrapping issue, but had to disable show/hide animations, as it was relying  on a specific height.

File: src/utils/newRelease.test.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    send: jest.fn(),
+  };
+
+  beforeEach(() => {
+    ipc.send.mockClear();
+    window.ipc = ipc;
+  });
+
+  it('Should return undefined if no ipc on window', () => {
+    delete window.ipc;
```
Comment (slaweet) [inline src/utils/newRelease.test.js]: `delete window.ipc;` should happen `afterEach` so that this test doesn't affect other (future) tests.

**Pull request merged**